packages/instantsearch-react/src/connectors/connectSearchBox.js

```diff
@@ -37,9 +37,16 @@ export default createConnector({
     return searchParameters.setQuery(getQuery(props, state));
   },
 
-  getMetadata(props) {
+  getMetadata(props, state) {
+    const query = getQuery(props, state);
     return {
       id: props.id,
+      filters: !query ? [] : [{
+        key: props.id,
+        label: query,
+        hide: true,
```

Comment:
Either solution sound valid 👍 . Let's move forward, and figure this out when we have a complete overview of the API?